File: 1013-FibonacciNumber/1013-FibonacciNumber.py
# Last updated: 04/04/2026, 23:32:30
class Solution(object):
    def fib(self, n):
        fiblist = [0,1]
        # 0 1 1 2 3 5 8 13 21 
        for i in range(n-1):
            fiblist.append(fiblist[-1]+fiblist[-2])
        return fiblist[n]


        

File: 1013-FibonacciNumber/1013-FibonacciNumber.java
// Last updated: 04/04/2026, 23:32:26
class Solution {
    public int fib(int n) {
        int[] memo = new int[n + 1];
        return computeFib(n, memo);
    }
    public static int computeFib(int n, int[] memo) {
        if (n <= 1) return n;
        if (memo[n] != 0) return memo[n];
        return memo[n] = computeFib(n - 1, memo) + computeFib(n - 2, memo);
    }
}
